frontend/src/components/Room.tsx

```diff
@@ -94,20 +103,67 @@ function Room() {
     } ));
 
 
+    const [roomAnnotation, setRoomAnnotation ] = useState(room?.roomAnnotation===null ? "" : String(room?.roomAnnotation))
+
+    const roomAnnotationChange = (e) => {
+        e.preventDefault();
+        pushNewRoomAnnotation.mutate(e.target.value)
+        setRoomAnnotation(e.target.value)
+    }
+
+    const pushNewRoomAnnotation = useMutation((newAnnotation) => {
+        const roomData = {
+            "roomAnnotation": newAnnotation
+        }
+
+        return Api.authApi.patch(`room/update/${id}`,roomData,{
+            headers: {
+                Authorization: `Bearer ${token}`,
+            },
+        })
+    })
+
+
+    useEffect( () => {
+        if (room?.roomAnnotation===null){
+            setRoomAnnotation("")
+        } else {
+            setRoomAnnotation(room?.roomAnnotation)
+        }
+    }, [room])
+
+
     return (
         <div className=" mx-16 py-10">
             <div className="flex flex-col items-center text-center">
                 <h1 className="text-3xl font-bold mb-4">Sala</h1>
                 {isRoomFetching ? (
                     <p className="text-center">Ładowanie danych...</p>
                 ) : (
-                    <div className="w-52 text-start">
+                    <div className="w-60 text-start">
                         <p className="pb-4 text-4xl text-center font-bold">{room?.number}</p>
 
-                        <p className="mb-2 text-lg">
-                            <span className="font-bold">Rodzaj: </span>
-                            {room?.annotation}
-                        </p>
+                        {!token ? (
+                            <p className="mb-2 text-lg">
+                                <span className="font-bold">Rodzaj: </span>
+                                {room?.roomAnnotation}
+                            </p>
+                        ) : (
+                            <div className="mb-2">
+                                <label className=" text-lg font-bold">Rodzaj:
+                                    <select value={roomAnnotation} onChange={roomAnnotationChange}
+                                            className="ml-1  font-normal text-center px-1 border-b-2 border-blue-500
+                                    hover:border-2 hover:border-blue-500 hover:rounded-md  focus:border-2
+                                    focus:border-blue-500 focus:rounded-md outline-none ">
+                                        <option value="laboratoryjna">laboratoryjna</option>
```

Comment:
couldn't values be stored as a enum, and here only reference to values? I think using enum might bebeneficial to change this value in one place